pkg/specgen/generate/namespaces.go

```diff
@@ -443,7 +447,11 @@ func specConfigureNamespaces(s *specgen.SpecGenerator, g *generate.Generator, rt
 		// the user or if we are creating a new UTS namespace.
 		// TODO: Should we be doing this for pod or container shared
 		// namespaces?
-		g.SetHostname(hostname)
+		if len(domainname) > 0 {
+			g.SetHostname(hostname + "." + domainname)
+		} else {
+			g.SetHostname(hostname)
+		}
```

Comment:
docker does not do this

Reply:
yeah you are right, I made a incorrect assumption about /etc/hostname. the  /etc/hosts behavior is correct though, I just need to add a default if domainname is specified and hostname is not